aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/data/AuthConfiguration.kt

```diff
@@ -20,77 +20,30 @@ import org.json.JSONObject
 /**
  * Configuration options for [AWSCognitoAuthPlugin].
  */
-data class AuthConfiguration internal constructor(val builder: Builder) {
-    val userPool: UserPoolConfiguration? = builder.userPool
-    val identityPool: IdentityPoolConfiguration? = builder.identityPool
+data class AuthConfiguration internal constructor(
+    val userPool: UserPoolConfiguration?,
+    val identityPool: IdentityPoolConfiguration?
+) {
 
     companion object {
         /**
-         * Returns a builder object for auth plugin configuration.
-         * @return fresh configuration builder instance.
-         */
-        @JvmStatic
-        fun builder(): Builder {
-            return Builder()
-        }
-
-        /**
-         * Returns a builder object populated from JSON.
-         * @return populated builder instance.
+         * Returns an AuthConfiguration instance from JSON
+         * @return populated AuthConfiguration instance.
          */
         internal fun fromJson(
             pluginJson: JSONObject,
-            configName: String = Config.DEFAULT.key
-        ): Builder {
-            return Builder(pluginJson, configName)
-        }
-
-        inline operator fun invoke(block: Builder.() -> Unit) = Builder().apply(block).build()
-    }
-
-    /**
-     * Builder class for constructing [AuthConfiguration].
-     */
-    class Builder constructor(
-        configJson: JSONObject? = null,
-        configName: String = Config.DEFAULT.key
-    ) {
-        var userPool: UserPoolConfiguration? = null
-        var identityPool: IdentityPoolConfiguration? = null
-
-        init {
-            configJson?.run {
-                optJSONObject(Config.COGNITO_USER_POOL.key)?.getJSONObject(configName)?.let {
-                    userPool = UserPoolConfiguration.fromJson(it).build()
-                }
-
-                optJSONObject(Config.CREDENTIALS_PROVIDER.key)?.getJSONObject(
-                    Config.COGNITO_IDENTITY.key
-                )
+            configName: String = "Default"
```

Comment:
We would want to support multiple configs, such as prod, beta, etc. Should we still have enum to represent this?